contracts/NonFungibleToken-v2.cdc

```diff
@@ -0,0 +1,158 @@
+to their Collection that only exposes the fields and functions in one or more
+of the interfaces. It also gives users the ability to make custom resources
+that implement these interfaces to do various things with the tokens.
+
+By using resources and interfaces, users of NFT smart contracts can send
+and receive tokens peer-to-peer, without having to interact with a central ledger
+smart contract.
+
+To send an NFT to another user, a user would simply withdraw the NFT
+from their Collection, then call the deposit function on another user's
+Collection to complete the transfer.
+
+*/
+
+import MetadataViews from "./MetadataViews.cdc"
+
+/// The main NFT contract interface. Other NFT contracts will
+/// import and implement this interface
+///
+pub contract NonFungibleToken {
+
+    /// Interface that the NFTs have to conform to
+    ///
+    pub resource interface NFT { //: MetadataViews.Resolver {
+        /// The unique ID that each NFT has
+        pub let id: UInt64
+
+        pub fun getViews(): [Type]
+        pub fun resolveView(_ view: Type): AnyStruct?
+    }
+
+    /// Interface to mediate withdraws from the Collection
+    ///
+    pub resource interface Provider {
+        /// withdraw removes an NFT from the collection and moves it to the caller
+        pub fun withdraw(withdrawID: UInt64): @AnyResource{NFT} {
+            post {
+                result.id == withdrawID: "The ID of the withdrawn token must be the same as the requested ID"
+            }
+        }
+    }
+
+    /// Interface to mediate withdraws from the Collection
+    ///
+    pub resource interface Transferable {
+        /// withdraw removes an NFT from the collection and moves it to the caller
+        pub fun transfer(id: UInt64, receiver: Capability<&AnyResource{Receiver}>): @AnyResource{NFT}
+    }
+
+    /// Interface to mediate deposits to the Collection
+    ///
+    pub resource interface Receiver {
+
+        /// deposit takes an NFT as an argument and adds it to the Collection
+        ///
+        pub fun deposit(token: @AnyResource{NFT})
+
+        /// getAcceptedTypes returns a list of NFT types that this receiver accepts
+        pub fun getAcceptedTypes(): [Type]
+    }
+
+    /// Interface that an account would commonly 
+    /// publish for their collection
+    pub resource interface CollectionPublic { //: MetadataViews.ResolverCollection {
+        pub fun deposit(token: @AnyResource{NFT})
+        pub fun borrowViewResolver(id: UInt64): &{MetadataViews.Resolver}
+        pub fun getIDs(): [UInt64]
+        pub fun borrowNFT(id: UInt64): &AnyResource{NFT}
+    }
+
+    /// Requirement for the concrete resource type
+    /// to be declared in the implementing contract
+    ///
+    pub resource interface Collection { //: Provider, Receiver, Transferable, CollectionPublic, MetadataViews.ResolverCollection {
+
+        /// Paths for the collection
+        pub let storagePath: StoragePath
```

Comment:
Personally I think the standard paths should be derived and not specified. If it includes the address it is deployed to and the name of the NFT then it is able to reason about it automatically.

Reply:
oh this would be very nice, also I suggest renaming those as `defaultStoragePath` maybe. This way it shows it is not forced but suggested.

Reply:
@bjartek Can you share how you would derive the standard paths?

Reply:
Get the identifier of the resouce type, normalize it (or allow "." in path) and use that. We do this for some things in .find